exchanges/orderbook/depth.go

```diff
@@ -21,48 +29,60 @@ type Depth struct {
 	alert.Notice
 
 	mux *dispatch.Mux
-	id  uuid.UUID
+	_ID uuid.UUID
 
 	options
+
+	// validationError defines current book state and why it was invalidated.
+	validationError error
+
 	m sync.Mutex
 }
 
 // NewDepth returns a new depth item
 func NewDepth(id uuid.UUID) *Depth {
 	return &Depth{
 		stack: newStack(),
-		id:    id,
+		_ID:   id,
 		mux:   service.Mux,
 	}
 }
 
 // Publish alerts any subscribed routines using a dispatch mux
 func (d *Depth) Publish() {
-	err := d.mux.Publish([]uuid.UUID{d.id}, d.Retrieve())
-	if err != nil {
+	if err := d.mux.Publish(d, d._ID); err != nil {
```

Comment:
I don't like this change. You're sending the actual orderbook. Yes, in RPC server, you are then calling `d.Retreive()`, but that puts trust in the end user to not impact the real orderbook.